apps/condo/domains/resident/schema/RegisterServiceConsumerService.js

```diff
@@ -61,12 +79,132 @@ const RegisterServiceConsumerService = new GQLCustomSchema('RegisterServiceConsu
             // TODO(pahaz): DOMA-4150 remove residentId and organizationId and migrate to ResidentWhereUniqueInput and OrganizationWhereUniqueInput
             type: 'input RegisterServiceConsumerInput { dv: Int!, sender: SenderFieldInput!, residentId: ID!, accountNumber: String!, organizationId: ID!, extra: RegisterServiceConsumerInputExtra }',
         },
+        {
+            access: true,
+            type: 'input RegisterResidentServiceConsumersInput { dv: Int!, sender: SenderFieldInput!, resident: ResidentWhereUniqueInput!, accountNumber: String!, paymentCategory: String }',
+        },
     ],
 
     mutations: [
         {
             doc: {
-                summary: 'Creates service consumer with default data, and automatically populates the optional data fields, such as `billingAccount',
+                summary: 'Creates service consumers with default data, and automatically populates the optional data fields, such as `billingAccount',
+                description: 'Search for accountNumbers across all organizations matching resident address and provided accountNumber. May return more than one ServiceConsumer in case of some organizations at resident address had same accountNumber. To be successfully created accountNumber and unitName should at least have billingAccount with same data or Meter with same data',
+                errors: ERRORS,
+            },
+            access: access.canRegisterServiceConsumer,
+            schema: 'registerResidentServiceConsumers(data: RegisterResidentServiceConsumersInput!): [ServiceConsumer!]',
+            resolver: async (_, args, context) => {
+                // TODO(DOMA-4138): remove paymentCategory usages
+                const { data: { dv, sender, resident: { id }, accountNumber, paymentCategory } } = args
+
+                if (!accountNumber || accountNumber.length === 0) throw new GQLError(ERRORS.ACCOUNT_NUMBER_IS_NOT_SPECIFIED, context)
+
+                const resident = await Resident.getOne(context, { id, deletedAt: null })
+                if (!resident) throw new GQLError(ERRORS.RESIDENT_NOT_FOUND, context)
+
+                const propertyWhere = resident.addressKey
+                    ? { addressKey: resident.addressKey }
+                    : { address_i: resident.address }
+
+                const residentProperties = await Property.getAll(context, propertyWhere)
+
+                let organizations = await Organization.getAll(context, {
+                    id_in: residentProperties.map(property => property.organization.id), deletedAt: null,
```

Comment:
Could there be several identical `organizations` here? If, for example, the `user` has several `residents` in different apartments of this `property`. If so, then it’s probably better to do `_.uniq(residentProperties.map(...))`